contracts/contracts/rem.swap/src/rem.swap.cpp

```diff
@@ -0,0 +1,329 @@
+/**
+ *  @copyright defined in eos/LICENSE.txt
+ */
+
+#include <rem.swap/rem.swap.hpp>
+
+#include "base58.cpp"
+#include "system_info.cpp"
+
+namespace eosio {
+
+   void swap::init( const name& rampayer, const string& txid, string& swap_pubkey,
+                    const asset& quantity, const string& return_address, const string& return_chain_id,
+                    const block_timestamp& swap_timestamp ) {
+      require_auth( rampayer );
+
+      const asset min_account_stake = get_min_account_stake();
+      asset producers_reward = get_producers_reward();
+
+      check_pubkey_prefix( swap_pubkey );
+      check( quantity.is_valid(), "invalid quantity" );
+      check( quantity.symbol == min_account_stake.symbol, "symbol precision mismatch" );
+      check( quantity.amount > min_account_stake.amount + producers_reward.amount, "the quantity must be greater "
+                                                                                   "than the swap fee" );
+      swap_pubkey = string( swap_pubkey.begin() + 3, swap_pubkey.end() );
+      const time_point swap_timepoint = swap_timestamp.to_time_point();
+
+      string swap_payload = join( { swap_pubkey, txid, remchain_id, quantity.to_string(), return_address,
+                                    return_chain_id, std::to_string( swap_timepoint.sec_since_epoch() ) } );
+
+      checksum256 swap_hash = hash( swap_payload );
+      swap_index swap_table( _self, get_first_receiver().value );
```

Comment:
store this variable as a member in class

Reply:
@Art17, done.

Reply:
@Art17, done.